core/esp_spi.c

```diff
@@ -269,7 +283,7 @@ size_t spi_transfer(uint8_t bus, const void *out_data, void *in_data, size_t len
         size_t offset = i * _SPI_BUF_SIZE;
         _spi_buf_transfer(bus, (const uint8_t *)out_data + offset,
             in_data ? (uint8_t *)in_data + offset : NULL, buf_size, e, word_size);
-        if (blocks) _rearm_extras_bit(bus, false);
+       _rearm_extras_bit(bus, false);
```

Comment:
`_rearm_extras_bit` function is to toggle off extras bit like address, dummy and command if data to send is higher than buffer.
The problem was only if the user use software CS. He need  manually toggle CS before send again ADDR/DUMMY/CMD . If the SPI device receive more data than buffer, he will receive theses bits as data.
This function is not needed if we are using hardware pin because esp toggle CS automatically.